extensions/interactions/CodeRepl/CodeRepl.js

```diff
@@ -100,16 +114,23 @@ oppia.directive('oppiaInteractiveCodeRepl', [
           });
         };
 
+        var fixLineNumbers = function(err) {
+          var preCodeNumLines = $scope.preCode.split('\n').length;
+          return err.replace(/on line ([0-9]+)/g, function(match, p1) {
+            return '(on line ' + String(Number(p1) - preCodeNumLines) + ')';
+          });
+        };
+
```

Comment:
This works as long as the error happenesd in the user's code, and not in the prepended or appended code. I have two ways in mind of solving this issue, One would be to show the prepended and appended code in the editor, but make these lines uneditable, similar to how the Javascript game "[untrusted](https://alexnisnevich.github.io/untrusted/)" does it. You have to get past the first level (which is very easy), to see it, so here's a screenshot:
![schermafdruk 2016-01-04 20 05 49](https://cloud.githubusercontent.com/assets/737493/12097931/f2ea9380-b31e-11e5-8e48-aeadab256912.png)

I suspect this won't work for all explorations though, because the answer to a question might be revealed in the code. I know this interaction provides a way to  give feedback based on an error, so I  would  not show the error, or show it in a limited way, if it is inside prepended or appended code, and rely on the exploration author to explain what went wrong based on the error. 
As an advanced addition to this, we might be able to notify the exploration author if there's an unexplained error. So if someone submits code that results in an error outside their input, and there's no explanation, we automatically post feedback. 

But for the purpose of this PR, I would just add a check for user-code boundaries, and say something like "Oh snap! an error occurred outside your code!"